src/hotspot/share/opto/parse2.cpp

```diff
@@ -2437,6 +2437,10 @@ void Parse::sharpen_type_after_if(BoolTest::mask btest,
             // at the control merge.
             _gvn.set_type_bottom(ccast);
             record_for_igvn(ccast);
+            if (tboth->is_inlinetypeptr()) {
+              assert(tboth->exact_klass(true)->is_inlinetype(), "");
```

Comment:
I don't think that assert is needed since `as_inline_klass` would already assert. And couldn't you use `tboth->inline_klass()`?